frictionless_ckan_mapper/ckan_to_frictionless.py

```diff
@@ -1,118 +1,123 @@
+        # `test_resource_path_is_set_even_for_uploaded_resources`
+        # is not expecting `url_type` in its output.
+        resource_keys_to_remove = ['package_id', 'position', 'url_type']
+
+        # TODO: delete keys last as may be needed for something in processing
+        for k in resource_keys_to_remove:
+            if k in resource:
+                del resource[k]
+        return resource
 
-    resource_mapping = {
-        'size': 'bytes',
-        'mimetype': 'mediatype',
-        'url': 'path'
-    }
+    @staticmethod
+    def _remap_resource_ckan_to_frictionless(resource):
+        '''Remap differences from CKAN to Frictionless resource'''
+        resource_mapping = {
+            'size': 'bytes',
+            'mimetype': 'mediatype',
+            'url': 'path'
+        }
 
-    # TODO: Do we want to keep everything except `package_id` and
-    # `position`? The current test
-    # `test_resource_path_is_set_even_for_uploaded_resources`
-    # is not expecting `url_type` in its output.
-    resource_keys_to_remove = ['package_id', 'position', 'url_type']
+        for key, value in resource_mapping.items():
+            if key in resource:
+                resource[value] = resource[key]
+                del resource[key]
+
+                # Cast CKAN resource size to int
+                if key == 'size':
+                    if not resource[value]:  # receives `null`
+                        resource[value] = 0
+                    else:
+                        resource[value] = int(resource[value])
+        return resource
 
     def resource(self, ckandict):
         '''Convert a CKAN resource to Frictionless Resource.
 
         1. Remove unneeded keys
         2. Expand extras.
-            * Extras are already expanded to key / values by CKAN (unlike on package)
-            * ~~Apply heuristic to unjsonify (if starts with [ or { unjsonify~~
-            * JSON loads everything and on error have a string
+            * Extras are already expanded to key / values by CKAN (unlike on
+            * package) ~~Apply heuristic to unjsonify (if starts with [ or {
+            * unjsonify~~ JSON loads everything and on error have a string
         3. Map keys from CKAN to Frictionless (and reformat if needed)
         4. Remove keys with null values (CKAN has a lot of null valued keys)
         4. Apply special formatting for key fields
         '''
-        # TODO: delete keys last as may be needed for something in processing
         resource = dict(ckandict)
-        for k in self.resource_keys_to_remove:
-            if k in resource:
-                del resource[k]
+        resource = self._resource_keys_to_remove(resource)
 
         # TODO: remove as CKAN Resource does not have extras (CKAN has already
-        # # unpacked them)
+        # unpacked them)
         if 'extras' in resource:
             resource['extras'] = json.loads(resource['extras'])
 
-        # unjsonify values
-        # 1. check if string
```

Comment:
I hope you don't mind, but I was encountering an unexpected parsing error with this comment left intact. According to Flake8, this looks good, but I wasn't able to remove the error without simplifying this comment.

https://www.flake8rules.com/rules/E265.html#:~:text=Block%20comment%20should%20start%20with%20'%23%20'%20(E265),%23%20)%20and%20the%20comment%20itself.